### PR TITLE
add support for draft styles

### PR DESCRIPTION
**Pull request by andrewharvey**
- [x] unit tests
- [x] real world tests to ensure I've covered every endpoints that supports draft, and none that don't actually support it.

### CODE REVIEW

File: services/styles.js [hunk 1]
```diff
@@ -368,22 +380,24 @@ Styles.getFontGlyphRange = function(config) {
...
     styleId: v.required(v.string),
     scrollZoom: v.boolean,
     title: v.boolean,
-    ownerId: v.string
+    ownerId: v.string,
+    draft: v.boolean
   })(config);
 
-  var fileName = config.styleId + '.html';
+  var fileName = config.styleId + (config.draft ? '/draft' : '') + '.html';
```
Comment (danswick) [inline services/styles.js]: When I was testing locally, the `/` in `/draft` was getting URI encoded. Do you think this is another case like https://github.com/mapbox/mapbox-sdk-js/pull/352#issuecomment-578498228?

Comment (andrewharvey) [reply services/styles.js]: oh good catch, you're right.

**Comment by andrewharvey** [comment]
I've resolved the merge conflicts and rebased after #356.

It would be appreciated if this could get some :eyes: to review, if it can be merged into master it would help prevent extra work resolving conflicts :wink:

**Comment by andrewharvey** [comment]
I'm not sure who's the maintainer of this project so I've requested a review from all those mentioned in https://github.com/mapbox/mapbox-sdk-js/pull/356

**Pull request merged**